cve_bin_tool/version_compare.py

```diff
@@ -0,0 +1,194 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import re
+
+"""
+A class for comparing arbitrary versions of products.
+
+Splits versions up using common whitespace delimiters and also splits out letters
+so that things like openSSL's 1.1.1y type of version will work too.
+
+This may need some additional smarts for stuff like "rc" or "beta" and potentially for
+things like distro versioning.  I don't know yet.
+"""
+
+
+class CannotParseVersionException(Exception):
+    """
+    Thrown if the version doesn't comply with our expectations
+    """
+
+
+def parse_version(version_string: str):
+    """
+    Splits a version string into an array for comparison.
+    This includes dealing with some letters.
+
+    e.g. 1.1.1a would become [1, 1, 1, a]
+    """
+    versionString = version_string
+    versionArray = []
+
+    # convert - and _ to be treated like . below
+    # we could switch to a re split but it seems to leave blanks so this is less hassle
+    versionString = versionString.replace("-", ".")
+    versionString = versionString.replace("_", ".")
+
```

Comment:
What about replacing all non-alphanumeric characters with '.' just in case?

Reply:
I opted not to do this because I'd rather have an error fall out so I can figure out how people are using other non-alphanum characters, but we definitely could do that.